juju/juju
# refactor apiserver/.../provisioner GetContainerProfileInfo for mocks, improve its test

## Description of change

GetContainerProfileInfo needed an improved test and to take advantage of container checking in the apiserver provisioner code.

## QA steps

1. juju bootstrap localhost
2. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-alt  --to lxd
3. juju add-unit lxd-profile-alt
4. juju add-machine lxd  

Check the charm profile was written correctly for both units and that the extra lxd container comes up with no issues.
